x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageOwnApiKeyClusterPrivilege.java

```diff
@@ -57,6 +57,15 @@ private ManageOwnClusterPermissionCheck() {
 
         @Override
         protected boolean extendedCheck(String action, TransportRequest request, Authentication authentication) {
+            if(authentication == null) {
+                //No valid credentials found
+                return false;
+            }
+            if (request == null) {
+                throw new IllegalArgumentException(
+                    "manage own cluster permission check only supported in context of an API key request");
+            }
+
```

Comment:
We should decide between returning `false` or throwing exceptions when the parameters are `null` or non-conformant, not both.

Although I would go with Exceptions because the permission in its natural authorization context shouldn't encounter such parameters (ie it's an unexpected condition), the permissions are checked in a stream and if one throws the others are not checked further, which is not what we want in the auditing context.

So my recommendation is:
```
if (request == null || authentication == null ) {
    // audit filtering may invoke this check method in a very loose way, and we shouldn't throw
    return false;
}
```